### PR TITLE
Add documentation for Interating a list using `zip` and `iter`

### PR DESCRIPTION
**Pull request by cwrw**

### CODE REVIEW

File: python/iterating_for_each_two_elements.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Interating a list for each 2 elements using `zip` and `iter`
+
+The `zip` function is used to return a list of tuples from each of the iterables passed to it
+
+```python
+  >>> list_one = ['a', 'b']
+  >>> list_two = ['c', 'd']
+  >>> zip(list_one, list_two)
```
Comment (mudge) [inline python/iterating_for_each_two_elements.md]: Is it also worth us showing what happens when you zip a list with _itself_ without using `iter` to show the different behaviour?

```python
>>> list = [1, 2, 3]
>>> zip(list, list)
[(1, 1), (2, 2), (3, 3)]
```

File: python/iterating_for_each_two_elements.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Interating a list for each 2 elements using `zip` and `iter`
```
Comment (mudge) [inline python/iterating_for_each_two_elements.md]: Perhaps there is a better way to phrase this (as we're not iterating over the resulting list)? "Converting a list into pairs of elements?"

What's the equivalent in Ruby? `each_slice(2)`?

**Comment by cwrw** [comment]
ah I intentionally broke it up as I thought it would make more sense. I'll change that

**Comment by mudge** [comment]
It definitely benefits from being broken down, I just wonder if it'd be easier to follow with something like:

* Introduction to `zip` for combining two lists
* Using `zip` on the same list twice
* Introduction to `iter` for converting lists into mutable iterators
* Using `zip` on the same `iter` twice

**Comment by cwrw** [comment]
hmmm I thought I was following that pattern, could you point me out to where I am not?

**Comment by mudge** [comment]
Update looks good but also includes the date range TIL?

**Comment by cwrw** [comment]
would you like me to open a new pr for that?

**Review by mudge — APPROVED** [review]

**Pull request merged**